apis/management.cattle.io/v3/backup_types.go

```diff
@@ -0,0 +1,64 @@
+package v3
+
+import (
+	"github.com/rancher/norman/condition"
+	"github.com/rancher/norman/types"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	BackupConditionCreated   condition.Cond = "Created"
+	BackupConditionCompleted condition.Cond = "Completed"
+	BackupConditionRestored  condition.Cond = "Restored"
+)
+
+type BackupTarget struct {
+	// s3 target
+	S3BackupTarget *S3BackupTarget `yaml:",omitempty" json:"s3BackupTarget"`
+}
+
+type S3BackupTarget struct {
+	// Access key ID
+	AccessKey string `yaml:"access_key" json:"accessKey,omitempty"`
+	// Secret access key
+	SecretKey string `yaml:"secret_key" json:"secretKey,omitempty" norman:"required,type=password" `
```

Comment:
@kinarashah will this be stored as a k8s secret out of the gate?

Reply:
Yeah, have a pr coming in with some minor changes, with that it should work out of the gate